File: laravel/public/clients/matlab/databaseList.m
% Displays the list of all visible Databases in the Ecosystem
%

serverURL='https://ecosystem.sonicom.eu/databases?type=json';

%% Fetch the list of databases from the Ecosystem
try
  options=weboptions; options.CertificateFilename=(''); 
  databases = webread(serverURL, options);
    % Check if the server returned a string instead of a struct/array.
  if ischar(databases)
    error('downloadFilesFromHTTPServer:serverError',...
          'Server returned a string, expected JSON.  Server response: %s', databases);
  end
catch ME
  error('downloadFilesFromHTTPServer:getFileList', ...
        'Failed to retrieve file list from server: %s.  Error: %s', serverURL, ME.message);
end

%% Check if correct JSON
if ~isstruct(databases) % check if structure
  error('downloadFilesFromHTTPServer:invalidFormat', 'Server did not return a struct of file information.');
end
if ~isfield(databases, 'data') % check if data in the structure
  error('downloadFilesFromHTTPServer:invalidFormat', 'Server did not return a JSON file information.');
end

%% Iterate through the database list and display each 
data=databases.data;
tab=struct2table(data);
disp(tab);

File: laravel/public/clients/python/databaseList.py
import requests

def databaseList():
    """
    Displays the list of all visible Databases in the Ecosystem.
    
    Author: Piotr Majdak (2025)
    
    """
    server_url = 'https://ecosystem.sonicom.eu/databases?type=json'

    # Fetch the list of databases from the Ecosystem
    try:
        response = requests.get(server_url)
        response.raise_for_status()  # Raise an HTTPError for bad responses (4xx or 5xx)
        databases = response.json()

    except requests.exceptions.RequestException as e:
        raise RuntimeError(f"Failed to retrieve file list from server: {server_url}. Error: {e}")

    # Check if correct JSON
    if not isinstance(databases, dict) or 'data' not in databases:
        raise ValueError("Server did not return a valid JSON structure with 'data' field.")

    # Iterate through the database list and display each
    data = databases['data']
    for item in data:
        database_id = item.get('ID')
        database_url = item.get('URL')
        database_title = item.get('Title')
        print(f"Database #ID {database_id}: {database_title}. URL: {database_url}")

if __name__ == "__main__":
    databaseList()